components/style/animation.rs

```diff
@@ -494,36 +502,48 @@ impl Animation {
             },
             _ => {},
         }
+    }
 
-        true
+    /// The length of the current iteration, a number between > 0 and <= 1, where
+    /// 1 is a full iteration and a number less than that represents an iteration that
+    /// is a fraction of the animation duration.
+    pub fn current_iteration_length(&self) -> f64 {
```

Comment:
This name is a bit odd because this is not quite a length (it's more of a... percentage? fraction?). But I don't think any of those would be particularly clearer...

Maybe `current_iteration_end_progress` or something?

Also may be worth adding:

```rust
fn current_iteration_duration(&self) -> f64 {
    self.current_iteration_end_progress() * self.duration
}
```

And change the two callers that do that themselves? Just a thought, if you disagree with these suggestions feel free to leave as is, I'm not 100% convinced of them myself :)

Reply:
I think these suggestions make sense, so I've incorporated them.